packages/typespec-client-generator-core/src/package.ts

```diff
@@ -364,6 +358,41 @@ function getServiceMethodLroMetadata(
       ) as SdkModelType,
     },
   };
+
+  function getFinalResponse(): SdkLroServiceFinalResponse | undefined {
+    if (
+      rawMetadata?.finalEnvelopeResult === undefined ||
+      rawMetadata.finalEnvelopeResult === "void"
+    ) {
+      return undefined;
+    }
+
+    const envelopeResult = diagnostics.pipe(
+      getClientTypeWithDiagnostics(context, rawMetadata.finalEnvelopeResult),
+    ) as SdkModelType;
+    const result = diagnostics.pipe(
+      getClientTypeWithDiagnostics(context, rawMetadata.finalResult as Model),
+    ) as SdkModelType;
+    const resultPath = rawMetadata.finalResultPath;
+    // find the property inside the envelope result using the final result path
+    let sdkProperty: SdkBodyModelPropertyType | undefined = undefined;
+    for (const property of envelopeResult.properties) {
+      if (property.__raw === undefined || property.kind !== "property") {
+        continue;
+      }
+      if (property.__raw?.name === resultPath) {
+        sdkProperty = property;
+        break;
+      }
```

Comment:
nit, personally I'd avoid `continue` in loop if possible. E.g. here we only need to reverse the condition of the `if`.